src/entity/Invite.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  Entity,
+  PrimaryColumn,
+  Column,
+  ManyToOne,
+  Index,
+  Generated,
+} from 'typeorm';
+import { User } from './User';
+import { FeatureType } from './Feature';
+
+export enum CampaignType {
+  Search = 'search',
+}
+
+@Entity()
+@Index('IDX_invite_userId_campaign', ['userId', 'campaign'], { unique: true })
+export class Invite {
+  @PrimaryColumn({ type: 'uuid' })
+  @Generated('uuid')
+  token: string;
+
+  @Column({ type: 'text' })
+  campaign: CampaignType | FeatureType;
+
+  @Column({ length: 36 })
+  userId: string;
+
+  @ManyToOne(() => User, {
+    lazy: true,
+    onDelete: 'CASCADE',
```

Comment:
My mistake, I have mentioned `cascade` before, but the right value should be 'SET NULL' as we wouldn't want to remove this row when the referenced entity is deleted.
```suggestion
    onDelete: 'SET NULL',
```

Reply:
is there any reason we want to keep `invite` records when a `user` is deleted? For reference, the feature table works the same way: https://github.com/dailydotdev/daily-api/blob/5f79df84b49df10824eee083c185270c103a7166/src/entity/Feature.ts#L23-L27

Reply:
another thing to consider is that if we allow setting `userId` to `NULL`, this will mess up our unique index on `userId` & `campaign`, since the moment a second user is deleted wich has an associated invite with the same campaign as the first, we would get a conflict

Reply:
Oh no. Nevermind my comment, I thought I was at the `Feature` entity. I checked and everything should be in order now. You are right, we want this deleted when the relevant user delets their account.